src/join/core.jl

```diff
@@ -274,6 +274,14 @@ function _innerjoin_unsorted(left::AbstractArray, right::AbstractArray{T}) where
     dict = Dict{T, Int}()
 
     right_len = length(right)
+    # we make sure that:
+    # * we do not preallocate dict to size larger than half of size of Int
+    #   (this is relevant in 32 bit architectures)
```

Comment:
```suggestion
    # * we do not preallocate dict of size larger than half of size of Int
    #   (this is relevant in 32 bit architectures)
```
(I don't want to imagine allocating a 2^63 length vector on 64-bit. :-D)